src/lib/registration/storage.rb

```diff
@@ -37,7 +38,10 @@ class RegCodes < Struct.new(:reg_codes)
 
       def initialize
         self.reg_codes = if Stage.initial
-          reg_codes_from_usb_stick || {}
+          raw_reg_code = Yast::Linuxrc.InstallInf("reg_code")
+          reg_code_from_install_inf = raw_reg_code ? [raw_reg_code.split(":", 2)].to_h : {}
```

Comment:
np: It would be nice to, perhaps, move these two lines to a separate method and document the format you expect.